jdrfandroidbleparser/src/test/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/characteristic/compoundcharacteristic/PnPIdCharacteristicTest.kt

```diff
@@ -111,7 +111,7 @@ class PnPIdCharacteristicTest : BaseTest() {
 
     @Test
     fun testProductId() {
-        val validPnPIdCharacteristic = PnPIdCharacteristic(mockBTCharacteristic(payload))
+        val validPnPIdCharacteristic = PnPIdCharacteristic(mockBTCharacteristic(pnpIdPayload))
```

Comment:
testProductId, testVendorId, testVendorIdSource can be folded into one test, as parsing the payload once should populated all three fields.

Reply:
I'm going to leave it as is because if we have one test that covers everything, and it breaks, we will have to go in and a debug the characteristic to see why it actually failed. By separating the tests out into three different tests, if the 'testProductId' test fails, but the others succeed, we know immediately, that a change happened that affected the ability to parse that individual field, and can address it directly.

There is also a test that checks if the successfulParse flag is set, and we test it by passing in both bad and good data. For most java projects I would agree with what you're saying, but in this case, we're pushing around bits... so explicit tests are going to help us narrow down any issues we run into when we have 100+ characteristics, vs 12.